### PR TITLE
feature/efr32_python_build_integration

### PR DESCRIPTION
**Pull request by lpbeliveau-silabs**
Insertion of build arguments in targets.py and efr32.py to make custom flags available to python build_examples.py script.

### CODE REVIEW

File: scripts/build/build/targets.py [hunk 1]
```diff
@@ -206,7 +206,20 @@ def BuildEfr32Target():
     target.AppendModifier('rpc', enable_rpcs=True)
     target.AppendModifier('with-ota-requestor', enable_ota_requestor=True)
     target.AppendModifier('sed', enable_sed=True)
-    target.AppendModifier('low-power', enable_low_power=True)
+    target.AppendModifier('low-power', enable_low_power=True).OnlyIfRe('-sed')
+    target.AppendModifier('chip_build_libshell', chip_build_libshell=True)
```
Comment (andy31415) [inline scripts/build/build/targets.py:210]: Consider simplifying suffixes. At least having `chip_` prefix everywhere seems to stutter.

How about `with_shell` or `shell` and `no_logging`  and such?

File: scripts/build/build/targets.py [hunk 2]
```diff
@@ -206,7 +206,20 @@ def BuildEfr32Target():
...
+    target.AppendModifier('disable_chip_logging', chip_logging=False)
+    target.AppendModifier('disable_chip_openthread_ftd', chip_openthread_ftd=False)
```
Comment (andy31415) [inline scripts/build/build/targets.py:212]: I would remove enable/disable from naming to make it simpler. `no_ftd` ? or does this mean mtd so should it be `mtd` ?

File: scripts/build/build/targets.py [hunk 3]
```diff
@@ -206,7 +206,20 @@ def BuildEfr32Target():
...
+    target.AppendModifier('enable_heap_monitoring', enable_heap_monitoring=True)
+    target.AppendModifier('disable_openthread_cli', enable_openthread_cli=False)
+    target.AppendModifier('show_qr_code', show_qr_code=True).ExceptIfRe('-low-power')
+    target.AppendModifier('wifi', enable_wifi=True)
+    target.AppendModifier('rs911x', enable_rs911x=True).OnlyIfRe('-wifi')
+    target.AppendModifier('wf200', enable_wf200=True).OnlyIfRe('-wifi')
+    target.AppendModifier('chip_enable_wifi_ipv4', enable_wifi_ipv4=True).OnlyIfRe('-wifi')
```
Comment (andy31415) [inline scripts/build/build/targets.py:219]: `wifi` seems sufficient

**Review by andy31415 — APPROVED** [review]

File: scripts/build/build/targets.py (hunk 1 above)
Comment (lpbeliveau-silabs) [reply scripts/build/build/targets.py]: agreed, names were changed to no_logging and shell.

File: scripts/build/build/targets.py (hunk 3 above)
Comment (lpbeliveau-silabs) [reply scripts/build/build/targets.py]: wifi_ipv4 replaced chip_enable_wifi_ipv4

File: scripts/build/build/targets.py (hunk 2 above)
Comment (lpbeliveau-silabs) [reply scripts/build/build/targets.py]: flag renamed to openthread_mtd, openthread was kept for more clarity.

**Review by jmartinez-silabs — APPROVED** [review]

File: scripts/build/builders/efr32.py [hunk 4]
```diff
@@ -137,6 +152,52 @@ def __init__(self,
...
+
+        if show_qr_code:
+            self.extra_gn_options.append('show_qr_code=true')
+
+        if enable_wifi:
+            self.dotfile += self.root + '/build_for_wifi_gnfile.gn'
+            if board == Efr32Board.BRD4161A:
+                self.extra_gn_options.append('is_debug=false chip_logging=false')
+            else:
+                self.extra_gn_options.append('disable_lcd=true use_external_flash=false')
```
Comment (jmeg-sfy) [inline scripts/build/builders/efr32.py:178]: i m not a fan of the disable_lcd , feel backward with all the rest of the enable_something convention

**Review by jmeg-sfy — APPROVED** [review]

File: scripts/build/builders/efr32.py (hunk 4 above)
Comment (lpbeliveau-silabs) [reply scripts/build/builders/efr32.py:178]: lcd is enabled by default in our configuration, only specific board require it disabled, which is why we chose that logic

**Pull request merged**